src/Page.js

```diff
@@ -861,6 +863,25 @@ Page.prototype.getPluginConfig = function () {
   };
 };
 
+/**
+ * Collect page sources provided by plugins
+ */
+Page.prototype.collectPluginSources = function (content) {
+  Object.entries(this.plugins).forEach(([pluginName, plugin]) => {
+    if (plugin.getSources) {
+      const sources = _.castArray(
+        plugin.getSources(content, this.pluginsContext[pluginName] || {},
+                          this.frontMatter, this.getPluginConfig()));
```

Comment:
Why `_.castArray`? `getSources` should return an array.

Reply:
So that a single source file could be returned as well,
should I enforce returning an array instead?

Reply:
Yes, since `getSources` is plural.